src/webcomponents/commons/tool-settings-update.js

```diff
@@ -0,0 +1,285 @@
+        this._toolSettings = UtilsNew.objectClone(this.toolSettings);
+        this.updatedFields = {};
+        this._config = {
+            ...this.getDefaultConfig(),
+            ...this.config,
+        };
+    }
+    // --- OBSERVERS ---
+    componentObserver() {
+        if (this.study && this.toolSettings && this.opencgaSession) {
+            this.#initOriginalObjects();
+        }
+    }
+
+    // --- EVENTS ---
+    onFieldChange(e) {
+        // The json has been modified, so we need to:
+        // FIXME: switch case
+        // 1. To update the local object settings where we store the settings modifications
+        if (e.detail.value?.json) {
+            this._toolSettings = {...e.detail.value?.json};
+        }
+        if (e.detail.param === "id") {
+            this._listStudies = e.detail.value.split(",");
+        }
+        // if (e.detail.param === "default") {debugger;}
+        // To notify that the json has been modified
+        // CAUTION 20230208 Vero: the toolSettings json has been updated
+        LitUtils.dispatchCustomEvent(this,
+            "studyToolSettingsUpdate",
+            null,
+            {_toolSettings: this._toolSettings}
+        );
+        this.requestUpdate();
+    }
+
+    onClear() {
+        NotificationUtils.dispatch(this, NotificationUtils.NOTIFY_CONFIRMATION, {
+            title: "Discard changes",
+            message: "Are you sure you want to discard the changes made?",
+            ok: () => {
+                this.#initOriginalObjects();
+                this.requestUpdate();
+            },
+        });
+    }
+
+    onSubmit() {
+        const toolName = this.#getResourceName(this.locus.toolId, "label");
+        // 1. Prepare query params
+        const params = {
+            includeResult: true,
+        };
+        // 2. Query
+        this.#setLoading(true);
+        this._listStudies.forEach(studyId => {
+            // 2.1. Get new study tool settings
+            const updateParams = OpencgaCatalogUtils.getNewToolIVASettings(this.opencgaSession, this.locus, this._toolSettings);
+            // 2.2 Query
+            this.opencgaSession.opencgaClient.studies()
+                // .update(this.opencgaSession.study.fqn, updateParams, params)
+                .update(studyId, updateParams, params)
+                .then(response => {
+                    // 1. Dispatch success notification
+                    NotificationUtils.dispatch(this, NotificationUtils.NOTIFY_SUCCESS, {
+                        title: `${toolName} Settings Update`,
+                        message: `${toolName} settings updated correctly`,
+                    });
+                    // 2. Dispatch study update event
+                    LitUtils.dispatchCustomEvent(
+                        this,
+                        "studyUpdateRequest",
+                        UtilsNew.objectClone(response.responses[0].results[0].fqn)
+                    );
```

Comment:
Please check if this is working as expected, because the `fqn` is a string not an object.